src/config/validator.c

```diff
@@ -183,4 +183,53 @@ bool validate_cidr(union conf_value *val, char err[VALIDATOR_ERRBUF_LEN])
 
 	free(str);
 	return true;
-}
\ No newline at end of file
+}
+
+// Validate IP address optionally followed by a port (separator is "#")
+bool validate_ip_port(union conf_value *val, char err[VALIDATOR_ERRBUF_LEN])
+{
+	// Check if it's a valid IP
+	char *str = strdup(val->s);
+	char *tmp = str;
+	char *ip = strsep(&tmp, "#");
+	char *port = strsep(&tmp, "#");
```

Comment:
This looks like a copy-paste error.
//suggestion
```
char *port = &tmp;
```

Reply:
No, this is correct. First, we make a copy of the string as `strsep()` modifies the string internally (it is setting null-terminators where delimiters are). We them store a "working" pointer `tmp` which is updated by the individual `strsep()` calls. The calls themselves return pointers to the string in between delimiters so exactly what we want.

Reply:
But there is only one`#` in `IP#Port`. How does the second one find anything?

Reply:
The first one if finds is `IP`. Then there is `#` separating both. The second thing it finds is `Port`.
If the string would be `IP#Port#` it would find three strings `IP`, `Host`, and a third empty one.

So this code is still correct, but your comment just revealed that we do not check if there is such a tail string, i.e., without having tested it, I guess `1.1.1.1#53#somecrap` would currently be accepted, too.

Reply:
Last commit:
```
$ sudo ./pihole-FTL --config dns.revServer.target "1.1.1.1#500#somecrap"
Invalid value: Not a valid IP ("1.1.1.1#500#somecrap")
```

Reply:
Glad I could help without understanding it ;-)

But I found the missing piece:

> In case no delimiter was found, the token
       is taken to be the entire string *stringp, and *stringp is made
       NULL.
       
https://man7.org/linux/man-pages/man3/strsep.3.html